packages/react-router/src/HeadContent.tsx

```diff
@@ -120,6 +120,23 @@ export const useTags = () => {
     structuralSharing: true as any,
   })
 
+  const styles = useRouterState({
+    select: (state) =>
+      (
+        state.matches
+          .map((match) => match.styles!)
+          .flat(1)
+          .filter(Boolean) as Array<RouterManagedTag>
+      ).map(({ children, ...style }) => ({
+        tag: 'style',
+        attrs: {
+          ...style,
+        },
```

Comment:
There is no need to make a shallow copy of `style` here, because it's already a *rest* object
```suggestion
      ).map(({ children, ...attrs }) => ({
        tag: 'style',
        attrs,
```